librz/bin/p/bin_gns1.c

```diff
@@ -0,0 +1,167 @@
+// SPDX-FileCopyrightText: 2025 Harsh Kumar <harsh237hk@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_types.h>
+#include <rz_util.h>
+#include <rz_lib.h>
+#include <rz_bin.h>
+
+// Apple C4000 (GNS1) Binary Format Loader
+
+#define GNS1_MAGIC     0x474E5331  // "GNS1" in hex
```

Comment:
I just checked the parser of other tools, and there is no magic, you need heuristic analysis (like mentioned in the issue) to automatically identify these files.